src/App.tsx

```diff
@@ -2,49 +1,109 @@
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.localeCompare(a) : a.localeCompare(b);
+        };
+
+        setActiveSort(SortType.Alphabetically);
+        break;
+
+      case SortType.ByLength:
+        compareFunction = (a: string, b: string) => {
+          return isReversed ? b.length - a.length : a.length - b.length;
+        };
+
+        setActiveSort(SortType.ByLength);
+        break;
+
+      default:
+        throw new Error(`Unknown sort type: ${typeOfSortGoods}`);
+    }
+
+    sortGoods(compareFunction);
+  };
+
+  const handleReverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevIsReversed => !prevIsReversed);
+  };
+
+  const handleResetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort('');
+  };
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-export const App: React.FC = () => {
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': activeSort !== SortType.Alphabetically,
+          })}
+          onClick={() => handleSort(SortType.Alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': activeSort !== SortType.ByLength,
+          })}
+          onClick={() => handleSort(SortType.ByLength)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', {
+            'is-warning': isReversed,
+            'is-warning is-light': !isReversed,
+          })}
+          onClick={handleReverseGoods}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(activeSort !== '' || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleResetGoods}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map((good, index) => (
+          <li data-cy="Good" key={index}>
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, you should use a unique identifier for the `key` prop instead of the index, if possible. This helps React identify which items have changed, are added, or are removed.